README.md

```diff
@@ -238,41 +240,43 @@ fruits=(Apple Pear Plum)
 
 ## Array slice
 
-Besides, we can take part of array using _slice_:
+Besides, we can extract a slice of array using the _slice_ operators:
 
 ```bash
 echo ${fruits[*]:0:2} # Apple Pear
 ```
 
-In example above, `fruits[*]` returns the content of array, `:0:2` takes a slice length of 2, starting at index 0.
+In the example above, `fruits[*]` returns the entire contents of the array, and `:0:2` extracts the slice of length 2, that starts at index 0.
 
-## Adding elements into array
+## Adding elements into an array
 
-If we wanna add new elements into array, we are happy, because it quite simple. Compound assignments come to help us here. We can use them such as:
+Adding elements into an array is quite simple too. Compound assignments are specially useful in this case. We can use them like this:
 
 ```bash
 fruits=(Orange ${fruits[*]} Banana Cherry)
-echo ${fruits[*]} # Orange Apple Pear Plum Banana Charry
+echo ${fruits[*]} # Orange Apple Pear Plum Banana Cherry
 ```
 
-## Deleting elements from array
+The example above, `fruits[*]` the entire contents of the array and substitutes it into the compound assignment, then assigns the new value into the `fruits` array mutating its original value.
+
+## Deleting elements from an array
 
-The `unset` command, which we are already familiar, is used to destroy arrays or element of array:
+To delete an element from an array, use the `unset` command:
 
 ```bash
 unset fruits[0]
-echo ${fruits[*]} #
+echo ${fruits[*]} # Apple Pear Plum Banana Cherry
```

Comment:
Nope, that can't be. Just run it your terminal. You have unset `fruits` before so it doesn't contain anything after that.